### PR TITLE
Remove explicit `files` and `outDir` values from tsconfig

### PR DESCRIPTION
**Pull request by apmorton**
When `files` is unspecified `input` defaults to `**/*`.
This pattern still only matches valid typescript extensions.

When `outDir` is specified it is implicitly added to `exclude`,
which takes priority over `include` and breaks the build.

### CODE REVIEW

**Review by supergrecko — APPROVED** [review]
LGTM! Maybe we should wait until the open typescript prs have been merged to reduce conflicts?

**Comment by mattgodbolt** [comment]
I merged all I can, so gonna merge

**Review by mattgodbolt — APPROVED** [review]

**Pull request merged**
